### ISSUE TITLE
prett-printy `Log Dictionary` option

### ISSUE THREAD

**Issue by oboehmer** [robotframework/robotframework#5082]
It would be good to have an option to pretty-print dictionaries, i.e. with indentation in the logs for better examination. Robot already pretty-prints dict return values in the log (when using TRACE option), so suggest to offer this also when doing it from keyword (maybe with an `indent=<int>` option which would then just do a json.dumps(dict, indent=indent) if set.

Happy to provide a PR to do this in 7.1

**Comment by JFoederer** [comment]
Doesn't the `formatter=repr` option from the [BuiltIn Log keyword](https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Log) already provide the requested feature?

_When using repr, bigger lists, dictionaries and other containers are also pretty-printed so that there is one item per row. For more details see [String representations](https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#String%20representations)._

**Comment by pekkaklarck** [comment]
Yeah, `formatter=repr` with the `Log` keyword ought to handle this. `Log Dictionary` also automatically logs items like:
```
key1: value1
key2: value2
```
I'm sure we could still enhance this, though. Is there something concrete you @oboehmer had in mind?

**Comment by oboehmer** [comment]
you are right, `formatter=repr` suits my case, closing this issue.
